modules/programs/hyprpaper.nix

```
@@ -0,0 +1,108 @@
+{ pkgs, config, lib, ... }:
+
+with lib;
+
+let cfg = config.programs.hyprpaper;
+in {
+  meta.maintainers = [ lib.maintainers.silmarp ];
+
+  options = {
+    programs.hyprpaper = {
+      enable = mkEnableOption "hyprpaper";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.hyprpaper;
+        defaultText = literalExpression "pkgs.hyprpaper";
+        description = "The package to use for the hyprpaper binary.";
+      };
+
+      systemd = {
+        enable =
+          mkEnableOption "Whether to enable hyprpaper systemd integration.";
+
+        target = mkOption {
+          type = types.str;
+          default = "graphical-session.target";
+          example = "hyprland-session.target";
+          description = ''
+            The systemd target that will automatically start the Hyprpaper service.
+          '';
+        };
+      };
+
+      settings = with types; {
+        preload = mkOption {
+          type = listOf path;
+          default = [ ];
+          description = ''
+            Wallpaper images that should be preloaded into memory
+          '';
+          example = [ ./wallpapers/tensura.png ];
+        };
+
+        wallpapers = mkOption {
+          type = listOf str;
+          default = [ ];
+          description = ''
+            Wallpaper to monitor mapper
+          '';
+          example = [ "monitor,./wallpapers/tensura.png" ];
+        };
+      };
+
+      extraConfig = mkOption {
+        type = types.str;
+        default = "";
+        description =
+          "Check https://github.com/hyprwm/hyprpaper#usage for info";
+        example = ''
+          newConfigOption = foo,bar
+        '';
+      };
+    };
+  };
+
+  config = {
+    home.packages = mkIf cfg.enable [ cfg.package ];
+
+    systemd.user.services.hyprpaper = mkIf cfg.systemd.enable {
+      Unit = {
+        Description = "autostart service for Hyprpaper";
+        Documentation = "https://github.com/hyprwm/hyprpaper";
+        BindsTo = [ "graphical-session.target" ];
+        After = [ "graphical-session-pre.target" ];
+      };
```

I guess this should restart when the configuration changes, if so you need something like

```suggestion
        X-Restart-Triggers = [ config."hypr/hyprpaper.conf".source ];
      };
```